lib/plugins/aws/package/compile/events/alb/index.js

```diff
@@ -28,8 +28,81 @@ class AwsCompileAlbEvents {
       },
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8020
-    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'alb', { type: 'object' });
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'alb', {
+      type: 'object',
+      properties: {
+        authorizer: { type: 'string' },
+        conditions: {
+          type: 'object',
+          properties: {
+            header: {
+              type: 'object',
+              properties: {
+                name: { type: 'string', maxLength: 40 },
+                values: { type: 'array', items: { type: 'string', maxLength: 128 } },
+              },
+              additionalProperties: false,
+              required: ['name', 'values'],
+            },
+            host: { type: 'string', pattern: '^[A-Za-z0-9*?.-]+$', maxLength: 128 },
+            ip: {
+              type: 'array',
+              items: {
+                oneOf: [
+                  { type: 'string', format: 'ipv4' },
+                  { type: 'string', format: 'ipv6' },
+                ],
+              },
+            },
+            method: {
+              type: 'array',
+              items: { type: 'string', pattern: '^[A-Za-z_-]+$', maxLength: 40 },
+            },
+            path: {
+              type: 'string',
+              pattern: '^([A-Za-z0-9*?_.$/~"\'@:+-]|&amp;)+$',
+              maxLength: 128,
+            },
+            query: {
+              type: 'object',
+              additionalProperties: { type: 'string', maxLength: 128 },
+              propertyNames: { type: 'string', maxLength: 128 },
+            },
+          },
+          required: ['path'],
+          additionalProperties: false,
+        },
+        healthCheck: {
+          oneOf: [
+            { type: 'boolean' },
+            {
+              type: 'object',
+              properties: {
+                healthyThresholdCount: { type: 'integer', minimum: 2, maximum: 10 },
+                intervalSeconds: { type: 'integer', minimum: 5, maximum: 300 },
+                matcher: {
+                  type: 'object',
+                  properties: {
+                    httpCode: { type: 'string' },
```

Comment:
We can probably restrict this with `pattern` as `^\d{3}(-\d{3})?(,\d{3}(-\d{3})?)*$`

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-targetgroup-matcher.html